llvm/lib/CodeGen/StackSlotColoring.cpp

```diff
@@ -67,6 +67,9 @@ namespace {
     const MachineBlockFrequencyInfo *MBFI = nullptr;
     SlotIndexes *Indexes = nullptr;
 
+    // - preserves Analysis passes in case RA may be called afterwards.
+    bool preserveRegAllocNeededAnalysis = false;
```

Comment:
There shouldn't be a reason to have a configuration flag for this. The pass can just unconditionally preserve the analyses.

This should also be a separate PR, not just a separate commit in this PR

Reply:
> There shouldn't be a reason to have a configuration flag for this. The pass can just unconditionally preserve the analyses.
> 
> This should also be a separate PR, not just a separate commit in this PR

ok sure, I should put changes in generic pass as separate PR.

But, as the same stackSlotColoring pass exists in generic way(after RA) for other Target's pipeline where these analysis are not really required to be preserved. As in this case, it just got additionally called in between AMDGPU target pipeline(where addtional analysis needs to be preserved). Therefore, got suggested to introduce this flag to avoid avoid any changes for other architecture's pipeline( maybe caused due to additional pass preserving in stackSlotColoring).

Reply:
Preserved does not mean required. Make them unconditionally preserved